### PR TITLE
Improve build scripts

### LINKED ISSUE #98: Run C++ tests in conda recipe
There are GTest tests but they are not started during conda package building.

### PR DESCRIPTION
**Pull request by PokhodenkoSA**
Fixes #98 but it would be better to move it to separate PR.

### CODE REVIEW

File: backends/CMakeLists.txt [hunk 1]
```diff
@@ -38,28 +38,13 @@ function (check_for_dpcpp)
...
-        message(STATUS "PYTHON_INCLUDE_DIR: ${PYTHON_INCLUDE_DIR}")
-    endif()
-endfunction()
-
 # Check for dpcpp in the specified DPCPP_ROOT
 check_for_dpcpp()
-# Check if the locations of Numpy and Python headers were provided
-check_for_numpy_inc()
+
+find_package(Python REQUIRED COMPONENTS Development NumPy)
```
Comment (diptorupd) [inline backends/CMakeLists.txt:44]: Do these macros search in the conda prefix?

File: backends/tests/CMakeLists.txt [hunk 2]
```diff
@@ -1,49 +1,40 @@
...
-
-  link_directories(${GTEST_LIB_DIR})
-
-  set(PYDPPL_BACKEND_TEST_CASES
-    test_sycl_kernel_interface
-    test_sycl_platform_interface
-    test_sycl_program_interface
-    test_sycl_queue_interface
-    test_sycl_queue_manager
+find_package(GTest REQUIRED)
```
Comment (diptorupd) [inline backends/tests/CMakeLists.txt:1]: Same question. I was not sure if the macro is smart enough to look under conda prefix.

File: conda-recipe/meta.yaml [hunk 3]
```diff
@@ -24,6 +24,7 @@ requirements:
         - make  # [unix]
         - ninja  # [win]
         - numpy >=1.17
+        - gtest
```
Comment (diptorupd) [inline conda-recipe/meta.yaml:27]: gtest 1.10 is needed as I am using GTEST_SKIP that only present in 1.10.

File: backends/CMakeLists.txt (hunk 1 above)
Comment (PokhodenkoSA) [reply backends/CMakeLists.txt:44]: Yes.

File: backends/tests/CMakeLists.txt (hunk 2 above)
Comment (PokhodenkoSA) [reply backends/tests/CMakeLists.txt:1]: I have tested on Linux. It works fine.

File: conda-recipe/meta.yaml (hunk 3 above)
Comment (PokhodenkoSA) [reply conda-recipe/meta.yaml:27]:
```suggestion
        - gtest >=1.10
```

File: backends/tests/CMakeLists.txt [hunk 4]
```diff
@@ -1,49 +1,40 @@
...
+
+include_directories(${GTEST_INCLUDE_DIR})
+include_directories(../include)
+
+link_directories(${GTEST_LIB_DIR})
+
+set(PYDPPL_BACKEND_TEST_CASES
+  test_sycl_kernel_interface
+  test_sycl_platform_interface
+  # test_sycl_program_interface
```
Comment (PokhodenkoSA) [inline backends/tests/CMakeLists.txt:20]: Note: commented.

File: backends/tests/CMakeLists.txt (hunk 2 above)
Comment (diptorupd) [reply backends/tests/CMakeLists.txt:1]: On my system I have `gtest` installed in my conda environment and `find_package` fails to locate Gtest. What do I need to do to fix? I can install `gtest` through `apt`, but that is not what we want right?

Comment (diptorupd) [reply backends/tests/CMakeLists.txt:1]: Ok. So to get it to work, I need `cmake` to come from conda as well. So it works. Nice!

**Comment by diptorupd** [comment]
Closing the PR as most of it is now implemented and the changes here are bit rotten beyond repair.

**Pull request closed without merging**